js/minesweeper.js

```diff
@@ -0,0 +1,210 @@
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function createField (width, height) {
+  var fieldArray = [];
+  for (var i = 0; i < height; i++) {
+    fieldArray[i] = [];
+    for (var j = 0; j < width; j++) {
+      fieldArray[i][j] = {
+        value: 0,
+        address: {
+          x: j,
+          y: i
+        },
+        checked: false, 
+        opened: false,
+        signed: false
+      };
+    }
+  }
+
+  return fieldArray;
+}
+
+function setMines (field, minesCount) {
+  var mines = 0;
+  while (mines !== minesCount) {
+    var minePositionX = getRandomInt(0, fieldWidth - 1);
+    var minePositionY = getRandomInt(0, fieldHeight - 1);
+    if (field[minePositionY][minePositionX].value !== "mine") {
+      field[minePositionY][minePositionX].value = "mine";
+      mines++;
+    }
+  }
+}
+
+function setCellsValues (array) {
+  for (var i = 0; i < array.length; i++) {
+    for (var j = 0; j < array[i].length; j++) {
+      if(array[i][j].value === "mine") {
+
+        continue;
+      }
+      var minesCount = 0;
+      for (var k = -1; k <= 1; k++) {
+        for (var l = -1; l <= 1; l++) {
+          if (array[i + k] && array[i + k][j + l] && array[i + k][j + l].value === "mine") {
```

Comment:
From first look it is hard to understand what is going on here - for such nested loops it is better to avoid short variable names like i, j, k etc. in this situation it would be better if k - row, and l - column, same for loop above.
Also this block contains 4 nested loops - it is difficult to see how it works. I guess it would be better to split this into two functions - then function name will explain what going on.